src/main/kotlin/com/tally/kiosk/common/base/BaseEntity.kt

```
@@ -0,0 +1,21 @@
+import jakarta.persistence.MappedSuperclass
+import org.springframework.data.annotation.CreatedBy
+import org.springframework.data.annotation.LastModifiedBy
+import org.springframework.data.jpa.domain.support.AuditingEntityListener
+
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener::class)
+abstract class BaseEntity protected constructor() : BaseTimeEntity() {
+    @CreatedBy
+    @Column(updatable = false)
+    lateinit var createdBy: String
+        protected set
```

detekt (1.23.7)

[warning] Usages of lateinit should be avoided.

(detekt.potential-bugs.LateinitUsage)